File: sdk/python/examples/parity.py
#!/usr/bin/env python3
"""Parity-пример: каноническая сводка четырёх вызовов SDK одной строкой JSON.

Кросс-языковое сравнение поведения SDK (sdk/CONTRACT.md): Rust и Java
parity-примеры печатают ту же строку. Запуск из корня монорепо:

    python3 sdk/python/examples/parity.py [КОРЕНЬ_РЕПО]
"""

import json
import sys
import tempfile
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1] / "src"))

from spine_be_sdk import SpineBE  # noqa: E402


def main() -> None:
    repo = Path(sys.argv[1]) if len(sys.argv) > 1 else Path(__file__).resolve().parents[3]
    fixtures = repo / "banking/demos/cli-from-claude-code"
    gate = fixtures / "scenario3-gate/fixtures"
    ir1 = fixtures / "scenario2-archify-cli/sbp-v1.architecture.json"
    ir2 = fixtures / "scenario2-archify-cli/sbp-v2.architecture.json"

    client = SpineBE()
    green = client.control_check(gate, constraints=gate / "CONSTRAINTS.yaml")
    validate = client.archify_validate("architecture", ir1)
    out = Path(tempfile.gettempdir()) / "sdk-parity-delta-py.html"
    compare = client.archify_compare(ir1, ir2, out)
    summary = compare["summary"]

    print(json.dumps({
        "green_passed": green.passed,
        "green_issues": len(green.issues),
        "validate_ok": validate["ok"],
        "checks": len(validate["checks"]),
        "compare_ok": compare["ok"],
        "comp_added": summary["components"]["added"],
        "conn_added": summary["connections"]["added"],
    }, ensure_ascii=False, separators=(",", ":")))


if __name__ == "__main__":
    main()

File: sdk/rust/examples/parity.rs
/// Parity-пример: каноническая сводка четырёх вызовов SDK одной строкой JSON.
/// Используется для кросс-языкового сравнения поведения SDK (sdk/CONTRACT.md).
use spine_be_sdk::{ArchifyReceipt, Client};
use std::path::PathBuf;

fn main() {
    let root = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("../..");
    let fixtures = root.join("banking/demos/cli-from-claude-code");
    let gate = fixtures.join("scenario3-gate/fixtures");
    let ir1 = fixtures.join("scenario2-archify-cli/sbp-v1.architecture.json");
    let ir2 = fixtures.join("scenario2-archify-cli/sbp-v2.architecture.json");

    let client = Client::new();
    let green = client
        .control_check(&gate, Some(&gate.join("CONSTRAINTS.yaml")))
        .expect("green control_check");
    let validate: ArchifyReceipt = client
        .archify_validate("architecture", &ir1)
        .expect("archify_validate");
    let tmp = std::env::temp_dir().join("sdk-parity-delta.html");
    let compare = client
        .archify_compare(&ir1, &ir2, &tmp)
        .expect("archify_compare");
    let summary = compare.summary().cloned().unwrap_or_default();

    println!(
        "{{\"green_passed\":{},\"green_issues\":{},\"validate_ok\":{},\"checks\":{},\"compare_ok\":{},\"comp_added\":{},\"conn_added\":{}}}",
        green.passed,
        green.issues.len(),
        validate.ok(),
        validate.checks().map_or(0, |c| c.len()),
        compare.ok(),
        summary["components"]["added"],
        summary["connections"]["added"],
    );
}
